agent/src/containeroptions/containerconfigparser.cpp

```diff
@@ -67,7 +70,7 @@ ContainerConfigParser::ContainerOptions ContainerConfigParser::parse(const std::
     size_t index;
     json_t *element;
 
-    ContainerOptions options;
+    DynamicContainerOptions options;
```

Comment:
It's a bit unusual in this component to pass a reference to the data to modify like this. Maybe this could be a member on the class instead?